### PR TITLE
Write configuration names that are valid for CMake

### LINKED ISSUE #520: Certain configuration names cause CMake errors
I converted an existing application to a CMakeLists.txt file using `Jucer2Reproducer`. The original application contained exporter configurations with these names:

````
Debug (32bit)
Debug (64bit)
Release (32bit)
Release (64bit)
````

The generated CMakeLists.txt thus contains lines like these:
````
jucer_export_target_configuration(
   "Visual Studio 2017"
   NAME "Debug (32bit)"
   DEBUG_MODE ON
   # ...
)
````

When running cmake I get these errors:
````
CMake Error at FRUT/prefix/FRUT/cmake/Reproducer.cmake:4550 (target_link_libraries):
   Error evaluating generator expression:
      $<CONFIG:Debug (32bit)>
   Expression syntax not recognized.
````

If I remove the spaces and brackets so that the names are `Debug32bit`. `Release64bit`, etc. the error disappears. It appears to me that `${config}` in `$<CONFIG:${config}>` must be escaped in some way but according to the CMake documentation, only angular brackets, commas and semicolons have proper escape sequences.

I would love to learn how to escape these properly. If that's not possible it would be a good solution to convert the incompatible configuration names to something that is compatible when converting the project with `Jucer2Reproducer`. 

I would be happy to contribute a fix for this, but at this point I'm unsure which option to choose.

### PR DESCRIPTION
**Pull request by McMartin**
Closes #520 and #521.

### CODE REVIEW

**Comment by McMartin** [comment]
@MartyLake: please review, thanks!

**Comment by McMartin** [comment]
@MartyLake gentle ping

File: Jucer2CMake/src/main.cpp [hunk 1]
```diff
@@ -181,6 +182,19 @@ juce::String makeValidIdentifier(juce::String s)
 }
 
 
+// Matches _FRUT_make_valid_configuration_name in Reprojucer.cmake.
+juce::String makeValidConfigurationName(const juce::String& configurationName)
+{
+  auto validName = configurationName.toStdString();
+
+  validName = std::regex_replace(validName, std::regex{"[^A-Za-z0-9_]+"}, " ");
```
Comment (MartyLake) [inline Jucer2CMake/src/main.cpp:190]: maybe extract regex already duplicated once ?

**Review by MartyLake — APPROVED** [review]
LGTM, with minor suggestion you can ignore :)

File: Jucer2CMake/src/main.cpp (hunk 1 above)
Comment (McMartin) [reply Jucer2CMake/src/main.cpp:190]: `^[A-Za-z0-9_]+$` and `[^A-Za-z0-9_]+` are not the same.

**Pull request merged**